src/CameraHandler.cpp

```diff
@@ -51,7 +54,50 @@ void CameraHandler::init(std::string uniqueId)
     mFramerateSet = false;
     mBuffersAllocated.clear();
     mStreamingNow.clear();
-    mCamera.reset(new Camera(uniqueId));
+
+    /*
+     * Determine whether the media pipeline needs to be configured the first for
+     * the video device by parsing "unique-id" property which is the following:
+     * unique-id = video-id[:media-id]
+     * where the "media-id" field is optional and should begin with ":"
+     */
+    std::string videoId, mediaId;
+    parseUniqueId(uniqueId, videoId, mediaId);
+
+    if (!mediaId.empty()) {
+        LOG(mLog, DEBUG) << "media-id is not empty, media pipeline needs to be configured";
+
+        mMediaController = MediaControllerPtr(new MediaController(mediaId));
+    }
+
+    if (!videoId.empty())
```

Comment:
Can it be that both videoID and mediaID configured by mistake?

Reply:
Theoretically yes, as everything in domain config, whether it is a display connector or block device number.  I have documented three possible options at the moment https://github.com/xen-troops/meta-xt-prod-devel/pull/253/commits/6081e0bd2e43b4170f5f0aa1afe8c4b6e49671d1#diff-2577518880b5e4b8f8d6b877d3042b5a for user to not dig into details too much what the proper configuration is, but choose needed camera type.